<launch>
  <!-- Set parameters for huno kinesthetics -->

  <!-- Joint Control Parameters -->
    <param name="/jointControl/joint_slew_limit" value="10.0" type="double" />
    <param name="/jointControl/loop_freq" value="20" type="double" />

    <param name="/kinematics/slew" value="6.0" type="double" />
    <rosparam param="/jointControl/min_limits">
    [112.0, 104.0, 204.0, 117.0, 113.0, 
     110.0, 117.0, 22.0, 104.0, 109.0, 
     75.0, 90.0, 60.0,
     120.0, 100.0, 110.0]</rosparam>
   <rosparam param="/jointControl/max_limits">
    [132.0, 124.0, 224.0, 137.0, 133.0,
     130.0, 137.0, 42.0, 124.0, 129.0,
     135.0, 180.0, 120.0,
     180.0, 160.0, 170.0]</rosparam>
    
  <!-- Huno Limb Parameters: Kinematics -->

  <!-- Left Leg -->
    <rosparam param="/leftFoot/min_thetas_C">
      [112.0, 104.0, 204.0, 117.0, 113.0] </rosparam>

    <rosparam param="/leftFoot/max_thetas_C">
      [132.0, 124.0, 224.0, 137.0, 133.0] </rosparam>

    <rosparam param="/leftFoot/thetaDotMaxs_C">
      [10.0, 10.0, 10.0, 10.0, 10.0] </rosparam>

    <rosparam param="/leftFoot/omegas">
      [-1,  0,  0,
        0,  1,  0,
        0,  1,  0,
        0, -1,  0,
        1,  0,  0] </rosparam>

    <rosparam param="/leftFoot/qs">
      [-0.06245,  0.0115, -0.0699,
       -0.02515,  0.0115, -0.0699,
        0.02385,  0.0115, -0.0699,
       -0.01545,  0.0115, -0.09515,
       -0.05375,  0.0115, -0.09515] </rosparam>

    <rosparam param="/leftFoot/thetaRefs">
      [127.6, 128.7, 225.7, 126.6, 126.6] </rosparam>

    <rosparam param="/leftFoot/g_0">
      [1, 0, 0, -0.05375,
       0, 1, 0,  0.0115,
       0, 0, 1, -0.10615,
       0, 0, 0,        1] </rosparam>

  <!-- Right Leg -->
    <rosparam param="/rightFoot/min_thetas_C">
      [110.0, 117.0, 22.0, 104.0, 109.0] </rosparam>

    <rosparam param="/rightFoot/max_thetas_C">
      [130.0, 137.0, 42.0, 124.0, 129.0] </rosparam> 

    <rosparam param="/rightFoot/thetaDotMaxs_C">
      [10.0, 10.0, 10.0, 10.0, 10.0] </rosparam>

    <rosparam param="/rightFoot/omegas">
      [-1,  0,  0,
        0, -1,  0,
        0, -1,  0,
        0,  1,  0,
       -1,  0,  0] </rosparam>

    <rosparam param="/rightFoot/qs">
      [-0.06245, -0.0115, -0.0699,
       -0.02515, -0.0115, -0.0699,
        0.02385, -0.0115, -0.0699,
       -0.01545, -0.0115, -0.09515,
       -0.05375, -0.0115, -0.09515] </rosparam>

    <rosparam param="/rightFoot/thetaRefs">
      [128.7, 128.7, 38.0, 130.8, 127.6] </rosparam>

    <rosparam param="/rightFoot/g_0">
      [1, 0, 0, -0.05375,
       0, 1, 0, -0.0115,
       0, 0, 1, -0.10615,
       0, 0, 0,        1] </rosparam>

  <!-- Left Arm -->
    <rosparam param="/leftHand/min_thetas_C">
      [75.0, 90.0, 60.0] </rosparam> 

    <rosparam param="/leftHand/max_thetas_C">
      [135.0, 180.0, 120.0] </rosparam>

    <rosparam param="/leftHand/thetaDotMaxs_C">
      [10.0, 10.0, 10.0] </rosparam>

    <rosparam param="/leftHand/omegas">
      [0, -1,  0,
       1,  0,  0,
       0,  1,  0] </rosparam>

    <rosparam param="/leftHand/qs">
      [-0.0263,   0.0292, -0.0266,
       -0.0263,   0.0672, -0.0266,
       -0.0263,   0.0672, -0.07575] </rosparam>

    <rosparam param="/leftHand/thetaRefs">
      [70.7, 38.0, 38.0] </rosparam>

    <rosparam param="/leftHand/g_0">
      [1, 0, 0,  0.0462,
       0, 1, 0,  0.0672,
       0, 0, 1, -0.07575,
       0, 0, 0,        1] </rosparam>

  <!-- Right Arm -->
    <rosparam param="/rightHand/min_thetas_C">
      [120.0, 100.0, 110.0] </rosparam>

    <rosparam param="/rightHand/max_thetas_C">
      [180.0, 160.0, 170.0] </rosparam>

    <rosparam param="/rightHand/thetaDotMaxs_C">
      [10.0, 10.0, 10.0] </rosparam>

    <rosparam param="/rightHand/omegas">
      [0,  1,  0,
       1,  0,  0,
       0, -1,  0] </rosparam>

    <rosparam param="/rightHand/qs">
      [-0.0263,  -0.0292, -0.0266,
       -0.0263,  -0.0672, -0.0266,
       -0.0263,  -0.0672, -0.07575] </rosparam>

    <rosparam param="/rightHand/thetaRefs">
      [187.7, 218.3, 221.5] </rosparam>

    <rosparam param="/rightHand/g_0">
      [1, 0, 0,  0.0462,
       0, 1, 0, -0.0672,
       0, 0, 1, -0.07575,
       0, 0, 0,        1] </rosparam>

</launch>
